docs/contributor/draft/rate-limit-api.md

```diff
@@ -0,0 +1,121 @@
+# Rate Limit Proposal
+
+## Spec 
+
+| field                                            | type        | description                                                     | required                      |
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| wokloadSelector                                  | object      | in case of empty, it's applied for the whole namespace          | no                            |
+| workloadSelector.labels                          | object      | allows to select given workload by a selector                   | no                            |
+| localRateLimit                                   | object      | allows to describe local rate limit properties                  | yes                           |
+| localRateLimit.by                                | object      | container for criteria you can limit by (for the future)        | yes                           |
+| localRateLimit.by.requests                       | object      | allows to set rate limit based on plain number of requests      | yes                           |
+| localRateLimit.by.requests.time                  | object      | allows to describe local rate limiting using time periods       | yes (if buckets not defined)  |
+| localRateLimit.by.requests.buckets               | object      | allows to describe local rate limiting using buckets            | yes (if time not defined)     |
+| localRateLimit.by.requests.time.unit             | string      | describes unit of time for rate limit                           | yes                           |
+| localRateLimit.by.requests.time.requestsLimit    | int         |describes requests limit for a declared time unit                | yes                           |
+| localRateLimit.by.requests.buckets.maxTokens     | int         |describes max amount of tokens that can stack up                 | yes                           |
+| localRateLimit.by.requests.buckets.tokensPerFill | int         | describes how many tokens to fill every defined period of time  | yes                           |
+| localRateLimit.by.requests.buckets.fillInterval  | time        |  describes how often fill tokens                                | yes                           |
+
+## Usage example
+
+```yaml
+apiVersion: operator.kyma-project.io/v1alpha1
```

Comment:
It would be good to also have an example to configure rate limit for ingress gateway, since those are basically the two scenarios where rate limit can be applied: on workload and on ingress gateway.